manifests/install.pp

```diff
@@ -69,18 +69,26 @@
   }
 
   $installDir = "${java::params::installRootDir}/jdk-${vendor}-${version}-${arch}"
+  
+  $major = inline_template('<%= scope.lookupvar(\'version\').split(\'-\')[0].gsub(\'.\', \'_\') %>')
+  # convert $major value to an integer
+  $majora = scanf($major, "%i") 
+  $majori = $majora[0]
```

Comment:
you don't test if `majora` is not an empty array. Is `$majora[0]` failing in that case ? or does it send an empty or nil value ?

Reply:
it would return an empty value. The validate_integer added on the last commit  will failed in this case.